packages/esm-patient-immunizations-app/src/immunizations/immunizations-detailed-summary.component.tsx

```diff
@@ -1,69 +1,67 @@
 import { mapFromFHIRImmunizationBundle } from './immunization-mapper';
 import { getImmunizationsConceptSet, performPatientImmunizationsSearch } from './immunizations.resource';
-import {
-  ImmunizationData,
-  ImmunizationSequenceDefinition,
-  ImmunizationWidgetConfigObject,
-  OpenmrsConcept,
-} from './immunization-domain';
+import first from 'lodash-es/first';
+import isEmpty from 'lodash-es/isEmpty';
+import DataTable, {
+  Table,
+  TableHead,
+  TableRow,
+  TableExpandHeader,
+  TableHeader,
+  TableBody,
+  TableExpandRow,
+  TableCell,
+  TableExpandedRow,
+} from 'carbon-components-react/es/components/DataTable';
+import Add16 from '@carbon/icons-react/es/add/16';
+import ImmunizationsForm from './immunizations-form.component';
+import Button from 'carbon-components-react/es/components/Button';
+import DataTableSkeleton from 'carbon-components-react/es/components/DataTableSkeleton';
+import SequenceTable from './immunizations-sequence-table';
+import { ExistingDoses, Immunization, Sequence } from '../types';
+import { findConfiguredSequences, findExistingDoses, latestFirst } from './utils';
 
 interface ImmunizationsDetailedSummaryProps {
   patient: fhir.Patient;
   patientUuid: string;
 }
 
-const ImmunizationsDetailedSummary: React.FC<ImmunizationsDetailedSummaryProps> = ({ patientUuid, patient }) => {
-  const config = useConfig();
-  const { t } = useTranslation();
-  const [allImmunizations, setAllImmunizations] = useState(null);
-  const [error, setError] = useState(null);
-  const immunizationsConfig: ImmunizationWidgetConfigObject = config.immunizationsConfig;
-
-  function findConfiguredSequences(configuredSequences: Array<ImmunizationSequenceDefinition>) {
-    return (immunizationsConceptSet: OpenmrsConcept): Array<ImmunizationData> => {
-      const immunizationConcepts: Array<OpenmrsConcept> = immunizationsConceptSet?.setMembers;
-      return map(immunizationConcepts, (immunizationConcept) => {
-        const immunizationDataFromConfig: ImmunizationData = {
-          vaccineName: immunizationConcept.display,
-          vaccineUuid: immunizationConcept.uuid,
-          existingDoses: [],
-        };
+enum StateTypes {
+  PENDING = 'pending',
+  RESOLVED = 'resolved',
+  ERROR = 'error',
+}
+interface ActionType {
+  type: 'pending' | 'resolved' | 'error';
+}
 
-        const matchingSequenceDef = find(
-          configuredSequences,
-          (sequencesDef) => sequencesDef.vaccineConceptUuid === immunizationConcept.uuid,
-        );
-        immunizationDataFromConfig.sequences = matchingSequenceDef?.sequences;
-        return immunizationDataFromConfig;
-      });
-    };
+const tableStatusReducer = (state: StateTypes, action: ActionType) => {
+  switch (action.type) {
+    case 'pending':
+      return StateTypes.PENDING;
+    case 'resolved':
+      return StateTypes.RESOLVED;
+    case 'error':
+      return StateTypes.ERROR;
+    default:
+      return StateTypes.PENDING;
   }
+};
```

Comment:
This could be simplified to a simple useState:

`const [status, setStatus] = useState(StateTypes.PENDING);`